automata/symbol/graph/symbol_graph.py

```diff
@@ -58,9 +48,8 @@ def __init__(
         Initializes a new instance of `SymbolGraph`.
         """
         super().__init__()
-        index = _load_index_protobuf(index_path)
```

Comment:
Let's actually keep this workflow, I think.

However, index_path should be `Optional` and similarly so should `index`. In the event that the graph is being loaded with 'STATIC', then `index` should be `None` when it is passed to the `GraphBuilder`.

The `GraphBuilder` then should check to confirm that `index` is not passed when loading with `from_pickle`. 

I could be missing a details here, but this is my initial thought.